src/openrct2/ride/Ride.cpp

```diff
@@ -3451,45 +3451,76 @@ static TrainReference VehicleCreateTrain(
     return train;
 }
 
-static bool VehicleCreateTrains(Ride& ride, const CoordsXYZ& trainsPos, TrackElement* trackElement)
+static bool VehicleCreateTrains(Ride& ride)
 {
     TrainReference firstTrain = {};
     TrainReference lastTrain = {};
-    int32_t remainingDistance = 0;
     bool allTrainsCreated = true;
 
-    for (int32_t vehicleIndex = 0; vehicleIndex < ride.NumTrains; vehicleIndex++)
+    std::vector<StationIndex> validStationIds;
```

Comment:
Consider using sfl::small_vector when there is no limit, with this you can specify enough static storage to avoid majority of heap allocs. If there is a maximum limit use sfl::static_vector.

Reply:
The hard limit is 255, though in practice it will likely be single digits. What would be appropriate, you think?

Reply:
Probably static_vector then, its just integers so it won't blow up the stack too much.